src/components/dialogs/contingency-list/contingency-list-utils.js

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import {
+    CONTINGENCY_LIST_TYPE,
+    CONTINGENCY_NAME,
+    COUNTRIES_1,
+    COUNTRIES_2,
+    EQUIPMENT_IDS,
+    EQUIPMENT_TABLE,
+    EQUIPMENT_TYPE,
+    ID,
+    NAME,
+    NOMINAL_VOLTAGE_1,
+    NOMINAL_VOLTAGE_2,
+    SCRIPT,
+} from '../../utils/field-constants';
+import {
+    ContingencyListTypeRefactor,
+    ElementType,
+} from '../../../utils/elementType';
+import { prepareContingencyListForBackend } from '../contingency-list-helper';
+import {
+    elementExists,
+    saveExplicitNamingContingencyList,
+    saveFormContingencyList,
+    saveScriptContingencyList,
+} from '../../../utils/rest-api';
+import yup from '../../utils/yup-config';
+import {
+    getRangeInputEmptyDataForm,
+    getRangeInputSchema,
+} from '../../utils/range-input';
+
+export const DEFAULT_TABLE_ROWS = [{}, {}, {}];
+
+export const CONTINGENCY_LIST_EQUIPMENTS = {
+    LINE: { id: 'LINE', label: 'Lines' },
+    TWO_WINDINGS_TRANSFORMER: {
+        id: 'TWO_WINDINGS_TRANSFORMER',
+        label: 'TwoWindingsTransformers',
+    },
+    GENERATOR: { id: 'GENERATOR', label: 'Generators' },
+    SHUNT_COMPENSATOR: { id: 'SHUNT_COMPENSATOR', label: 'ShuntCompensators' },
+    HVDC_LINE: { id: 'HVDC_LINE', label: 'HvdcLines' },
+    BUSBAR_SECTION: { id: 'BUSBAR_SECTION', label: 'BusBarSections' },
+    DANGLING_LINE: { id: 'DANGLING_LINE', label: 'DanglingLines' },
+};
+const checkNameIsUnique = (name, activeDirectory) => {
+    if (!name) {
+        return false;
+    }
+    return new Promise((resolve) => {
+        elementExists(activeDirectory, name, ElementType.CONTINGENCY_LIST).then(
+            (val) => resolve(!val)
+        );
+    });
+};
+
+export const getSchema = (activeDirectory) => {
+    return yup.object().shape({
+        [NAME]: yup
```

Comment:
I'm not sure it cover all case of NameWrapper component.
Maybe use the component instead ?